ui/plugin-system/src/components/OptionsEditorTabs/OptionsEditorTabs.tsx

```diff
@@ -25,10 +25,14 @@ export type OptionsEditorTab = {
 
 export type OptionsEditorTabsProps = {
   tabs: OptionsEditorTab[];
+  mode?: 'Add' | 'Edit';
 };
 
-export const OptionsEditorTabs = ({ tabs }: OptionsEditorTabsProps) => {
-  const [activeTab, setActiveTab] = useState(0);
+export const OptionsEditorTabs = ({ tabs, mode = 'Add' }: OptionsEditorTabsProps) => {
+  // If user is adding a panel, default to first tab ("General")
+  // If user is editing a panel, default to second tab ("Query" or "Markdown")
+  const defaultTab = mode === 'Add' ? 0 : 1;
+  const [activeTab, setActiveTab] = useState(defaultTab);
```

Comment:
I'm wondering if this logic would live better up in the `PanelSpecEditor` and passed down to a `defaultTab` prop, so that if it gets more complex in the future, it lives a little closer to panel configuration?

Reply:
I like that, and we can make the tabs themselves a bit "dumber"